sdk/core/System.ClientModel/src/Message/BinaryContent.cs

```diff
@@ -191,4 +200,47 @@ public override void Dispose()
             }
         }
     }
+
+    private sealed class StreamBinaryContent : BinaryContent
+    {
+        private readonly Stream _stream;
+        private readonly long _origin;
+
+        public StreamBinaryContent(Stream stream)
+        {
+            if (!stream.CanSeek)
+            {
+                throw new ArgumentException("Stream must be seekable.", nameof(stream));
+            }
+
+            _stream = stream;
+            _origin = stream.Position;
+        }
+
+        public override bool TryComputeLength(out long length)
+        {
+            // CanSeek is validated in constructor - it will always be true.
```

Comment:
this is an edge case, but `CanSeek` can return false after the stream is disposed.

Reply:
Thank you @christothes!  I will reevaluate this in light of your comment -- I remember now you mentioning this in the past, but I had forgotten.  Thank you!